compiler/qsc_frontend/src/lower.rs

```diff
@@ -921,10 +934,14 @@ impl With<'_> {
     }
 }
 
-fn lower_callable_kind(kind: ast::CallableKind) -> hir::CallableKind {
-    match kind {
-        ast::CallableKind::Function => hir::CallableKind::Function,
-        ast::CallableKind::Operation => hir::CallableKind::Operation,
+fn lower_callable_kind(kind: ast::CallableKind, attrs: &[qsc_hir::hir::Attr]) -> hir::CallableKind {
+    if attrs.contains(&qsc_hir::hir::Attr::Measurement) {
+        hir::CallableKind::Measurement
+    } else {
+        match kind {
+            ast::CallableKind::Function => hir::CallableKind::Function,
+            ast::CallableKind::Operation => hir::CallableKind::Operation,
+        }
     }
```

Comment:
It just occurred to me that this logic makes it so that a function can have the `@Measurement` attribute and it will still work, which doesn't really make sense. I think this points to two options:
 - Update this helper function to be able to produce errors (it would take `self` and callers would be updated, so that `self.lowerer.errors.push` can be used as it is elsewhere).
 - Keep just the original two `CallableKind` and update the logic in partial eval and the HIR pass to peek at the attributes to get the info about whether this callable was marked as a measurement.

I think the first option would be fine, just wanted to mention the second as a comparison. Either way, worth adding a unit test to the right spot to confirm that a function with `@Measurement` produces an error.

Reply:
Good catch. Fixed.